### PR TITLE
[WIP] Update hybrid model parallel GPT2 example for up to 175B parameters

### PR DESCRIPTION
**Pull request by kurisusnowdeng**
- Added scripts for ColossalAI, Megatron-Deepspeed and Pytorch FSDP

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,3 +2,7 @@
 	path = inference
 	url = https://github.com/hpcaitech/EnergonAI.git
 	branch = main
+[submodule "examples/language/gpt/experiments/hybrid_parallel/megatron_deepspeed"]
+	path = examples/language/gpt/experiments/hybrid_parallel/megatron_deepspeed
+	url = https://github.com/kurisusnowdeng/Megatron-LM.git
+	branch = deepspeed
```
Comment (FrankLeeeee) [inline .gitmodules:8]: why do we need this?

**Comment by FrankLeeeee** [comment]
@kurisusnowdeng is this PR still valid? If not, I will close it.

**Comment by kurisusnowdeng** [comment]
> @kurisusnowdeng is this PR still valid? If not, I will close it.

@FrankLeeeee We're still working on this PR. Basically some evaluations are left, including gemini & 8bit optimizer pretraining performance, as well as unitests.

**Comment by FrankLeeeee** [comment]
> 

Ok, I will convert this PR to draft for now as commits will trigger the CI repeatedly.

**Comment by kurisusnowdeng** [comment]
> > 
> 
> Ok, I will convert this PR to draft for now as commits will trigger the CI repeatedly. @kurisusnowdeng do make it ready for review once it is ready :)

Thanks a lot

**Pull request closed without merging**